### DISCUSSION TITLE
How long are magic links valid for?

### DISCUSSION THREAD

**Discussion by mstibbard** [supabase/supabase#2367] [Questions] [answered]
I am testing out magic links in dev currently and my link expired. Is there a specific duration the link is valid for? And if so, could we specify the duration to the user in the email template that is sent?

I could not find an answer in the documentation (please point me to it if I am wrong here!).

**Comment by mcmx** [comment, marked as answer]
This is handled by the gotrue API in the function **recoverVerify** in **verify.go** file, the value is hardcoded for 24 hours, also the token for signup verification is valid for 24 hours.


https://github.com/supabase/gotrue/blob/master/api/verify.go#L196

**Reply by mstibbard** [reply]
Thanks! That's great to know - and should be added to the documentation (both that it is 24 hours AND that it is not configurable).
